v-next/hardhat-utils/src/sub-process.ts

```diff
@@ -0,0 +1,27 @@
+import { spawn } from "node:child_process";
+
+/**
+ * Spawns a detached subprocess to execute a given file with optional arguments.
+ *
+ * @param absolutePathToSubProcessFile - The absolute path to the subprocess file to be executed.
+ * @param [args=[]] - Optional arguments to pass to the subprocess.
+ *
+ * The subprocess runs in a detached mode and its standard input/output is ignored.
+ * This function does not wait for the subprocess to complete and the subprocess is unreferenced
+ * to allow the parent process to exit independently.
+ */
+export function spawnSubProcess(
+  absolutePathToSubProcessFile: string,
+  args: string[] = [],
+): void {
+  const subprocess = spawn(
+    process.execPath,
+    ["--import", "tsx/esm", absolutePathToSubProcessFile, ...args],
+    {
+      detached: true,
```

Comment:
Do we want to allow to pass ENV variables too via the "env" option?

Reply:
Yes, I think we'd need it.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5471/commits/870d0e5d59f95aa9a4ff872dd8498604706dc721)